docs/design/provisioning.md

```diff
@@ -0,0 +1,143 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Abstract
+
```

Comment:
Could we add a bit more context? For example, it would be helpful to state the use case:
"This proposal is intended to provide a solution that will enable an administrator to have a single method to expose all the storage available in the kubernetes via an arbitrary classification system. For example, an administrator that wants to provide access to Cinder, Ceph RBD and NFS Shares to the constituents of their Kubernetes cluster can choose to abstract the description of the implementations of the available storage systems of into arbitrary classifications like Gold, Silver and Bronze. The constituents of the Kubernetes cluster only need to know about about classifications are available and what they provide."

I will add that controlling access to which tiers are available to which users is intrinsic to the overall use case, but I don't want to overly burden this specific proposal with that requirement unless there is no path forward to add it.